docs/physics/montecarlo/initialization.ipynb

```diff
@@ -0,0 +1,289 @@
+    "$$\\Delta t = \\frac{1}{L}=\\frac{1}{4 \\pi R_{\\mathrm{phot}}^2 \\sigma_{\\mathrm{R}} T_{\\mathrm{phot}}^4}.$$\n",
+    "\n",
+    "During packet initialization, each packet is assigned an initial propagation direction $\\mu$ which is the cosine of the angle $\\theta$ which the packet's path makes with the radial direction. Using a pseudo-random number generator which generates numbers $z$ uniformly distributed on the interval $[0,1]$, the propagation direction is determined according to\n",
+    "$$\\mu = \\sqrt{z}.$$\n",
+    "This sampling is demonstrated in the code below.\n",
+    "\n",
+    "Finally, each packet is assigned an initial frequency (or more precisely, the initial frequency of its consitiuent photons). Note that since each packet has the same energy, each packet will represent a different number of real photons. The sampling on packet frequencies is more involved than that of the propagation direction, as it involves sampling the Planck distribution (see below). TARDIS uses the technique described in <strong data-cite=\"Carter1975\">[]</strong> and summarized in <strong data-cite=\"Bjorkman2001\">[]</strong> for this purpose.\n",
+    "\n",
+    "During the simulation, the energy of the packet remains constant in the local\n",
+    "co-moving frame (see [Reference Frames](propagation.rst#reference-frames)). This naturally ensures energy\n",
+    "conservation and constitutes the main advantage of this discretization scheme. **However, while the energy of the packets is conserved in the co-moving frame, the frequency of the contituent photons (in the local co-moving frame) may vary over the course of the simulation. Thus, a packet may represent several different numbers of real photons throughout their lifetimes.**\n",
+    "\n",
+    "We now demonstrate the TARDIS packet initialization framework:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "426325e5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from tardis.montecarlo.packet_source import *\n",
+    "from tardis import constants as const\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "#The random number generator that will be used\n",
+    "rng = np.random.default_rng()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4ae02998",
+   "metadata": {},
+   "source": [
+    "The following cell contains values that you can change to see how it affects the spectrum:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bc34bf33",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "seed = 1 #seed for the pseudo-random number generator\n",
+    "r_phot = 1e15 #radius of the supernova's photosphere in cm\n",
+    "n_packets = 40000 #number of packets generated"
```

Comment:
This cell and the next few code cells have some minor PEP 8 violations. In this cell, the inline comments do not follow the guidelines: "Inline comments should be separated by at least two spaces from the statement. They should start with a # and a single space."
Maybe reformat this cell and the next few using black outside of the notebook?